### ISSUE TITLE
Opening project on Ubuntu or macOS Catalina print errors mentioning Viewport and nullptr; did the same once on Win10 too

### ISSUE THREAD

**Issue by sszigeti** [godotengine/godot#55937]
### Godot version

3.4.stable + 3.4.1-rc2

### System information

Windows 10, Linux (Ubuntu), macOS 10.15.7

### Issue description

Opening my game project has suddenly started outputting a lot of errors, but the game can still works properly. 

I only saw these errors on Windows once, but on Ubuntu I see them every time and cannot suppress them in any way.

```
   sandor@Legion:~$ Godot/Godot_v3.4.1-rc2_x11.64 
   Godot Engine v3.4.1.rc2.official.d5aa00c2c - https://godotengine.org
   OpenGL ES 3.0 Renderer: GeForce GTX 1050 Ti/PCIe/SSE2
   OpenGL ES Batching: ON
   /lib/x86_64-linux-gnu/libasound.so.2: undefined symbol: snd_dlpath
   
   Editing project: /home/sandor/GameDev/GameProjectName/GameCodeName (::home::sandor::GameDev::GameProjectName::GameCodeName)
   Godot Engine v3.4.1.rc2.official.d5aa00c2c - https://godotengine.org
   OpenGL ES 2.0 Renderer: GeForce GTX 1050 Ti/PCIe/SSE2
   OpenGL ES Batching: ON
   /lib/x86_64-linux-gnu/libasound.so.2: undefined symbol: snd_dlpath
   
   sandor@Legion:~$ ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: (Node not found: "Viewport" (relative to "/root/EditorNode/@@592/@@593/@@601/@@603/@@607/@@611/@@612/@@613/@@629/@@630/@@639/@@640/@@6279/@@6113/@@6114/@@6115/@@6116/@@6117/Bootstrap").)
      at: get_node (scene/main/node.cpp:1325)
   ERROR: ViewportTexture: Path to node is invalid.
      at: setup_local_to_scene (scene/main/viewport.cpp:69)
   ERROR: Condition "!cc" is true. Returned: nullptr
      at: get_current_version (drivers/gles2/shader_gles2.cpp:216)
   ERROR: Condition "!version" is true. Returned: false
      at: bind (drivers/gles2/shader_gles2.cpp:87)
   ERROR: Condition "!version" is true. Returned: -1
      at: _get_uniform (drivers/gles2/shader_gles2.h:258)
   ERROR: Condition "!version" is true. Returned: -1
      at: _get_uniform (drivers/gles2/shader_gles2.h:258)
   ERROR: Condition "!version" is true. Returned: -1
      at: _get_uniform (drivers/gles2/shader_gles2.h:258)
   ERROR: Condition "!version" is true. Returned: -1
      at: _get_uniform (drivers/gles2/shader_gles2.h:258)
   ERROR: Condition "!version" is true. Returned: -1
      at: _get_uniform (drivers/gles2/shader_gles2.h:258)
   ERROR: Condition "!version" is true. Returned: -1
      at: _get_uniform (drivers/gles2/shader_gles2.h:258)
```

The "Bootstrap" scene is the main scene, a super simple data loader scene that displays a "loading" message then proceeds to a main menu, replacing this scene. It has no `tool` script.

The game is using GLES2, but switching to GLES3 didn't make any difference, the errors were still there.

### Steps to reproduce

Unsure.

I've asked for ideas on Reddit to circumvent issues stemming from `yield(get_tree(), "idle_frame")`, and after [reverting a failed experiment Godot 3.4 on Windows has suddenly froze](https://www.reddit.com/r/godot/comments/rg462n/ideas_for_replacing_yieldget_tree_idle_frame/hoiz7yu/). 

I then moved my `.import` directory elsewhere (essentially deleting it from the project) but Godot's loading window remained empty white. Finally I had to use v3.3.3 to re-create the imports. After that the project opens again properly even in 3.4 on Windows, but on Linux I see the errors whenever I open the project for editing.

**Comment by sszigeti** [comment]
Just tested with macOS 10.15.7 (a few hours ofter Apple has kindly fixed their OS), and the game is working without issues on v3.4.stable. Even after starting Godot from the Terminal there are no errors whatsoever.

**Comment by akien-mga** [comment]
> The "Bootstrap" scene is the main scene, a super simple data loader scene that displays a "loading" message then proceeds to a main menu, replacing this scene. It has no `tool` script.

I understand there are no `tool` scripts in the Bootstrap scene, but are there `tool` scripts otherwise in the project? The error does look like something is messing up with the `EditorNode` (which is the editor's main scene).

There might be corruption of some editor-specific metadata, you can try to clear `~/.config/godot/projects/<projectname-hash>` and see if that solves it. (Back it up first in case that's indeed the problem, so that we can inspect the contents.)

**Comment by sszigeti** [comment]
@akien-mga Thank you for the guidance! After moving my `~/.config/godot/projects/<projectname-hash>` elsewhere I still saw the same errors on Ubuntu. On the Mac my `~/.config` doesn't have a `godot` directory in it, in fact I couldn't find anything with `find ~ -path '*/godot'` at all.

However I do have a total of 10 `tool` scripts in the project, so I should be able to try and see what happens if check each of them. I'll report my findings back here in a day or two.

**Comment by sszigeti** [comment]
I started removing the `tool` keywords from my scripts, one by one. I didn't expect anything to break as these were only used for updating exported textures/colors in the editor, which would happen in the game anyways. The mentioned errors persisted, even after the last trace of `tool` was removed.

Then, after opening the game project in v3.3.3 I saw that everything was still working and the error messages were gone (as they have never been present in that Godot version). Then returning to v3.4.1-rc2 the error messages haven't appeared either. It's pretty strange, I'm not sure how I should reproduce the issue now, but I'll try it, and record my experiences here.

**Comment by sszigeti** [comment]
Suddenly I started seeing 2 of the below error message, though my game remained editable and could compile and run it without issues.
```
ERROR: Index current_edited_scene = -1 is out of bounds (edited_scene.size() = 0).
   at: get_edited_scene_root (editor/editor_data.cpp:648)
```
I have [discovered](https://www.reddit.com/r/godot/comments/rjt9sa/error_index_current_edited_scene_1_is_out_of/) that these were actually caused by Viewport textures, directly connected to a TextureRect. The temporary fix was to connect the viewport textures only in the TextureRect's `_ready` function, though previously this was working, and I haven't touched these parts of the code for months.

I have also removed all the `tool` scripts, and also factored out all `setget` accessors, so I no longer see the reported issues, therefore I'm closing this issue report, as I'm no longer in the position to investigate further.
